components/eamxx/tests/uncoupled/mam4/input.yaml

```diff
@@ -9,8 +9,8 @@ time_stepping:
   number_of_steps: ${NUM_STEPS}
 
 atmosphere_processes:
-  atm_procs_list: (MAMMicrophysics)
-  MAMMicrophysics:
+  atm_procs_list: (mam4microphysics)
+  mam4_microphysics:
```

Comment:
Should these match `mam4_micro`, or are they different for a reason?

Reply:
yeah, this does seem like a bug. The name within the parentheses must match verbatim the name on the next line.

Reply:
The real issue is that we have names for `AtmosphereProcess`s that are independent of how these processes are registered in `register_physics`. This issue is only going to cause more confusion as time passes.

For now, I'm going to change all these names to be `mam4_micro`, but in my view the right way to solve this is to enforce that the registered name and the `name()` method are always the same.

Reply:
Ideally, we would like to call `P3::name()` in the factory registration, but the `name` method is not static (and cannot be, since it's virtual).

We could have derived classes have a "name_static()" method, and have their `name()` function call the static one, and use the static method in the factory (to avoid bugs due to typos). Unfortunately we _do_ need the name method, since the base class uses it when it's printing messages inside generic interfaces (e.g., put the ap name in the logger, or in the timings file).